server/src/core/evaluation.rs

```diff
@@ -562,27 +566,47 @@ impl Evaluation {
                 }
                 let mut context = Some(base_eval[0].symbol.context.clone());
                 //TODO context should give params
-                let (base_sym_ref, instance) = base_eval[0].symbol.get_symbol(session, &mut context, &mut diagnostics, None);
-                let base_sym = base_sym_ref.upgrade();
+                let base_sym_weak_eval= base_eval[0].symbol.get_symbol(session, &mut context, &mut diagnostics, None);
+                let base_sym = base_sym_weak_eval.weak.upgrade();
                 if let Some(base_sym) = base_sym {
                     if base_sym.borrow().typ() == SymType::CLASS {
-                        if instance {
+                        if matches!(base_sym_weak_eval.symbol_type, EvaluationSymbolType::Instance) {
                             //TODO handle call on class instance
                         } else {
-                            //TODO diagnostic __new__ call parameters
-                            evals.push(Evaluation{
-                                symbol: EvaluationSymbol {
-                                    sym: EvaluationSymbolPtr::WEAK(EvaluationSymbolWeak{
-                                        weak: base_sym_ref.clone(),
-                                        instance: true,
-                                    }),
-                                    context: HashMap::new(),
-                                    factory: None,
-                                    get_symbol_hook: None,
-                                },
-                                value: None,
-                                range: Some(expr.range)
-                            });
+                            if base_sym.borrow().get_tree() == (vec![S!("builtins")], vec![S!("super")]){
+                                match parent.borrow().get_in_parents(&vec![SymType::CLASS], true){
+                                    None => (), // TODO, diagnostic? or just leave empty
+                                    Some(parent_class) =>
```

Comment:
You only handle super(), and not super(Class) nor super(Class, instance), right?